src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1734,6 +1734,154 @@ Literal Literal::sha512(NodeStorage &node_storage) const {
     return this->hash_with("SHA2-512", node_storage);
 }
 
+Literal Literal::now(NodeStorage &node_storage) {
+    auto n = std::chrono::floor<std::chrono::milliseconds>(std::chrono::system_clock::now());
+    datatypes::registry::Timezone tz{};
+    datatypes::registry::TimePoint t = tz.to_local(n);
+    datatypes::registry::OptionalTimezone opt = std::nullopt;
+    return make_typed_from_value<datatypes::xsd::DateTime>(std::make_pair(t, opt), node_storage);
+}
+
+Literal Literal::year(NodeStorage &node_storage) const {
```

Comment:
These functions should all be called `as_*`, i.e. `as_year`. And there should be an additional set of functions called the way these are called which return an int (uint?) directly.

Also: noexcept